library/psa_crypto.c

```diff
@@ -545,6 +545,68 @@ static inline size_t psa_get_key_slot_bits( const psa_key_slot_t *slot )
     return( slot->attr.bits );
 }
 
+/** Return the output MAC length of a MAC algorithm, in bytes
+ *
+ * \param[in] algorithm     The specific MAC algorithm
+ * \param[in] key_type      The key type of the key to be used with the
+ *                          \p algorithm.
+ * \param[out] length       The calculated output length of the given MAC
+ *                          \p algorithm when used with a key corresponding to
+ *                          the given \p key_type
+ *
+ * \retval #PSA_SUCCESS
+ *         The \p length has been successfully calculated
+ * \retval #PSA_ERROR_NOT_SUPPORTED
+ *         \p algorithm is a MAC algorithm, but ubsupported by this PSA core.
+ * \retval #PSA_ERROR_INVALID_ARGUMENT
+ *         \p algorithm is not a valid, specific MAC algorithm or \p key_type
+ *         describes a key incompatible with the specified \p algorithm.
+ */
+static psa_status_t psa_get_mac_output_length( psa_algorithm_t algorithm,
```

Comment:
It would be good to declare this function `MBEDTLS_STATIC_TESTABLE` and replicate the `PSA_MAC_LENGTH` test in `test_suite_psa_crypto_metadata.function`. This way we'd have positive unit tests for basically free.

Reply:
Done